packages/common-all/src/store/ConfigStore.ts

```diff
@@ -0,0 +1,336 @@
+import { errAsync, okAsync, ResultAsync } from "neverthrow";
+import { DendronError, IDendronError } from "../error";
+import { URI, Utils } from "vscode-uri";
+import { ConfigReadOpts, IConfigStore } from "./IConfigStore";
+import { IFileStore } from "./IFileStore";
+import { CONSTANTS, StatusCodes } from "../constants";
+import { ConfigUtils, DeepPartial } from "../utils";
+import { DendronConfig, DendronConfigValue, DVault } from "../types";
+import { fromStr, toStr } from "../yaml";
+import { DConfigLegacy } from "../oneoff/ConfigCompat";
+import _ from "lodash";
+
+export class ConfigStore implements IConfigStore {
+  private _fileStore: IFileStore;
+  private _wsRoot: URI;
+  private _cachedDendronConfig: DendronConfig | undefined;
+  private _homeDir: URI | undefined;
+
+  get path(): URI {
+    return Utils.joinPath(this._wsRoot, CONSTANTS.DENDRON_CONFIG_FILE);
+  }
+
+  constructor(fileStore: IFileStore, wsRoot: URI, homeDir: URI | undefined) {
+    this._fileStore = fileStore;
+    this._wsRoot = wsRoot;
+    this._homeDir = homeDir;
+  }
+
+  create(
+    defaults?: DeepPartial<DendronConfig>
+  ): ResultAsync<DendronConfig, IDendronError> {
+    const config: DendronConfig = ConfigUtils.genLatestConfig(defaults);
+
+    const configDumpResult = toStr(config);
+    if (configDumpResult.isErr()) {
+      return errAsync(configDumpResult.error);
+    }
+
+    const configDump = configDumpResult.value;
+    return ResultAsync.fromPromise(
+      Promise.resolve(
+        this._fileStore.write(this.path, configDump).then((resp) => {
+          // TODO: remove throws and chain results instead.
+          // until IFileStore methods are changed to use neverthrow
+          // instead of RespV3, we need to manually throw here
+          // in order for this ResultAsync to properly handle it
+          if (resp.error) {
+            throw resp.error;
+          }
+          return config;
+        })
+      ),
+      (err) => err as DendronError
+    );
+  }
+
+  readRaw(): ResultAsync<
+    DeepPartial<DendronConfig>,
+    IDendronError<StatusCodes | undefined>
+  > {
+    return ResultAsync.fromPromise(
+      Promise.resolve(
+        this._fileStore.read(this.path).then((resp) => {
+          // TODO: remove throws and chain results instead.
+          // until IFileStore methods are changed to use neverthrow
+          // instead of RespV3, we need to manually throw here
+          // in order for this ResultAsync to properly handle it
+          if (resp.error) {
+            throw resp.error;
+          }
+
+          const readPayload = resp.data;
+
+          const configLoadResult = fromStr(readPayload);
+          if (configLoadResult.isErr()) {
+            throw configLoadResult.error;
+          }
+          return configLoadResult.value as DeepPartial<DendronConfig>;
+        })
+      ),
+      (err) => err as DendronError
+    );
```

Comment:
You could simplify this to:
```suggestion
          returnresp.data;
      ),
      (err) => err as DendronError
    ).andThen((readPayload) => {
          return fromStr(readPayload);
        })
    )
```